### DISCUSSION TITLE
connect a linphone account with baresip

### DISCUSSION THREAD

**Discussion by nikhilkoottungal** [baresip/baresip#3005] [Q&A]
Hi Team,
I have compiled the baresip source code for my laptop and I am trying to register it with a linphone sip account -> "sip:[njolly@sip.linphone.org](mailto:njolly@sip.linphone.org)".
I tried adding this in the baresip using the command - "/uanew sip:njolly:[10Johxxxxxx@sip.linphone.org](mailto:10Johxxxxxx@sip.linphone.org) sip.linphone.org".
while trying this out I get the following error :
'reg: sip:[njolly@sip.linphone.org](mailto:njolly@sip.linphone.org) (prio 0): Register: Permission denied [13]'.

I also tried adding the User Agent in the ~/.baresip/account file as following this format: 'sip:njolly@sip.linphone.org;transport=tcp;auth_pass=secret'

The linphone account which i am trying to add is verified using the linphone app in android devices.

Can some one suggest a proper method to add the above mentioned linphone id, in the baresip.

**Comment by juha-h** [comment]
So you have not tried with `;auth_user=something`?

**Reply by nikhilkoottungal** [reply]
hi @juha-h ,
Thanks for the response. I tried adding ' sip:njolly@sip.linphone.org;transport=tcp;auth_user=njolly;auth_pass=10Joxxxx', in the accounts file.
and here is the log catpured after running baresip

'baresip v3.10.1 Copyright (C) 2010 - 2024 Alfred E. Heggestad et al.
Local network addresses:
  wlp0s20f3:  192.168.1.17
     virbr0:  192.168.122.1
  wlp0s20f3:  2401:4900:1cdc:2ec8:1ec2:b72d:ae5b:1f6
  wlp0s20f3:  2401:4900:1cdc:2ec8:3423:77a1:c455:b829
  wlp0s20f3:  fe80::1e0b:b5c3:840d:942c
aucodec: PCMU/8000/1
aucodec: PCMA/8000/1
aufilt: auconv
aufilt: auresamp
ausrc: alsa
auplay: alsa
medianat: stun
medianat: turn
medianat: ice
Populated 1 account
Populated 6 contacts
Populated 2 audio codecs
Populated 2 audio filters
Populated 0 video codecs
Populated 0 video filters
baresip is ready.
**reg: sip:njolly@sip.linphone.org (prio 0): Register: Function not implemented [38]
reg: sip:njolly@sip.linphone.org (prio 0): Register: Function not implemented [38]
reg: sip:njolly@sip.linphone.org (prio 0): Register: Function not implemented [38]
reg: sip:njolly@sip.linphone.org (prio 0): Register: Function not implemented [38]
reg: sip:njolly@sip.linphone.org (prio 0): Register: Function not implemented [38]
reg: sip:njolly@sip.linphone.org (prio 0): Register: Function not implemented [38]
reg: sip:njolly@sip.linphone.org (prio 0): Register: Function not implemented [38]**


I also tried '/uanew sip:njolly:[10Johxxxxxx@sip.linphone.org](mailto:10Johxxxxxx@sip.linphone.org) sip.linphone.org"', which yielded another error as i mentioned:
'**reg: sip:[njolly@sip.linphone.org](mailto:njolly@sip.linphone.org) (prio 0): Register: Permission denied [13]**'.

Kindly let  me know if any more information is needed. I would really appreciate the support on this.

**Comment by juha-h** [comment]
Start baresip with -s option.  Then you see, if REGISTER requests is sent out and if so, what the response is.

**Reply by nikhilkoottungal** [reply]
baresip v3.10.1 Copyright (C) 2010 - 2024 Alfred E. Heggestad et al.
Local network addresses:
  wlp0s20f3:  192.168.1.17
     virbr0:  192.168.122.1
  wlp0s20f3:  2401:4900:1cdc:2ec8:1ec2:b72d:ae5b:1f6
  wlp0s20f3:  2401:4900:1cdc:2ec8:3423:77a1:c455:b829
  wlp0s20f3:  fe80::1e0b:b5c3:840d:942c
aucodec: PCMU/8000/1
aucodec: PCMA/8000/1
aufilt: auconv
aufilt: auresamp
ausrc: alsa
auplay: alsa
medianat: stun
medianat: turn
medianat: ice
Populated 1 account
Populated 6 contacts
Populated 2 audio codecs
Populated 2 audio filters
Populated 0 video codecs
Populated 0 video filters
baresip is ready.
20:49:35.592#
TLS 192.168.1.17:48944 -> 147.135.128.132:5223
REGISTER sip:sip.linphone.org SIP/2.0
Via: SIP/2.0/TLS 192.168.1.17:48944;branch=z9hG4bK68996fcabbff11fc;rport
Contact: <sip:njolly-0x55ec0ba721c0@192.168.1.17:48944;transport=tls>;expires=3643;+sip.instance="<urn:uuid:f3fdf6fc-d23d-c567-d898-eb6f6a86c7a0>"
Max-Forwards: 70
To: <sip:njolly@sip.linphone.org>
From: <sip:njolly@sip.linphone.org>;tag=61a9b6c44901e495
Call-ID: c67f934829ea562e
CSeq: 64157 REGISTER
User-Agent: baresip v3.10.1 (x86_64/Linux)
Allow: INVITE,ACK,BYE,CANCEL,OPTIONS,NOTIFY,SUBSCRIBE,INFO,MESSAGE,UPDATE,REFER
Content-Length: 0


20:49:36.000#
TLS 147.135.128.132:5223 -> 192.168.1.17:48944
SIP/2.0 401 Unauthorized
Via: SIP/2.0/TLS 192.168.1.17:48944;branch=z9hG4bK68996fcabbff11fc;rport=25083;received=106.222.237.183
From: <sip:njolly@sip.linphone.org>;tag=61a9b6c44901e495
To: <sip:njolly@sip.linphone.org>;tag=U4Ur9Fj21876a
Call-ID: c67f934829ea562e
CSeq: 64157 REGISTER
Server: Flexisip/2.3.3-33-g3ded5934 (sofia-sip-nta/2.0)
WWW-Authenticate: Digest realm="sip.linphone.org", nonce="BwrN6QAAAAA59RAbAAB6aqNdICcAAAAA", opaque="+GNywA==", algorithm=SHA-256, qop="auth"
Content-Length: 0


reg: sip:njolly@sip.linphone.org (prio 0): Register: Function not implemented [38]

**Comment by juha-h** [comment]
Ans leave out `;transport=tcp`.

**Reply by nikhilkoottungal** [reply]
please find the logs after removing 'transport=tcp'
sip:njolly@sip.linphone.org;auth_user=njolly;auth_pass=10Joxx;

baresip is ready.
20:54:51.044#
TLS 192.168.1.17:43584 -> 147.135.128.132:5223
REGISTER sip:sip.linphone.org SIP/2.0
Via: SIP/2.0/TLS 192.168.1.17:43584;branch=z9hG4bKb5b026de38c239eb;rport
Contact: <sip:njolly-0x562a9d62b1c0@192.168.1.17:43584;transport=tls>;expires=3819;+sip.instance="<urn:uuid:f3fdf6fc-d23d-c567-d898-eb6f6a86c7a0>"
Max-Forwards: 70
To: <sip:njolly@sip.linphone.org>
From: <sip:njolly@sip.linphone.org>;tag=4acb70747ca063f9
Call-ID: 9148678d83079717
CSeq: 7118 REGISTER
User-Agent: baresip v3.10.1 (x86_64/Linux)
Allow: INVITE,ACK,BYE,CANCEL,OPTIONS,NOTIFY,SUBSCRIBE,INFO,MESSAGE,UPDATE,REFER
Content-Length: 0


20:54:51.489#
TLS 147.135.128.132:5223 -> 192.168.1.17:43584
SIP/2.0 401 Unauthorized
Via: SIP/2.0/TLS 192.168.1.17:43584;branch=z9hG4bKb5b026de38c239eb;rport=1789;received=106.222.237.183
From: <sip:njolly@sip.linphone.org>;tag=4acb70747ca063f9
To: <sip:njolly@sip.linphone.org>;tag=D88NZ7F3Scegm
Call-ID: 9148678d83079717
CSeq: 7118 REGISTER
Server: Flexisip/2.3.3-33-g3ded5934 (sofia-sip-nta/2.0)
WWW-Authenticate: Digest realm="sip.linphone.org", nonce="QwvN6QAAAACPuTyTAADW8zsi5BEAAAAA", opaque="+GNywA==", algorithm=SHA-256, qop="auth"
Content-Length: 0


reg: sip:njolly@sip.linphone.org (prio 0): Register: Function not implemented [38]

**Comment by juha-h** [comment]
nikhilkoottungal writes:

> WWW-Authenticate: Digest realm="sip.linphone.org", nonce="QwvN6QAAAACPuTyTAADW8zsi5BEAAAAA", opaque="+GNywA==", algorithm=SHA-256, qop="auth"
> Content-Length: 0
> 
> reg: ***@***.*** (prio 0): Register: Function not implemented [38]

Could have something to do with  opaque="+GNywA==" or algorithm=SHA-256.

**Reply by nikhilkoottungal** [reply]
@juha-h , No I am using default configurations only. I havent made any changes related to 'opaque="+GNywA==" or algorithm=SHA-256.
If this seems to be an issue, could you suggest any configuration changes to to fix it?

**Comment by juha-h** [comment]
Looks like baresip does not support SHA-256, but only md5:
```
static bool auth_handler(const struct sip_hdr *hdr, const struct sip_msg *msg,
			 void *arg)
{
        ...
	if (pl_isset(&ch.algorithm) && pl_strcasecmp(&ch.algorithm, "md5")) {
		err = ENOSYS;
		goto out;
	}
```
Open an issue about it.

**Reply by nikhilkoottungal** [reply]
thanks @juha-h for the reply. will open the issue

**Reply by nikhilkoottungal** [reply]
@juha-h Can you please suggest any sip servers, which can be used to test the functionality of  baresip for now. 
Also any test sip accounts which can be  used temporarily will help in this case.

**Comment by juha-h** [comment]
nikhilkoottungal writes:

> @juha-h Can you please suggest any sip servers, which can be used to
> test the functionality of  baresip for now.

Try sip2sip.info.

**Reply by nikhilkoottungal** [reply]
Thanks a lot...Tested with sip2sip.info

**Comment by cspiel1** [comment]
> Thanks for the response. I tried adding ' sip:njolly@sip.linphone.org;transport=tcp;auth_user=njolly;auth_pass=10Joxxxx', in the accounts file.

I am pretty sure that this is not a correct format. It should be:
`<sip:njolly@sip.linphone.org;transport=tcp>;auth_user=njolly;auth_pass=10Joxxxx`

See https://github.com/baresip/baresip/wiki/Accounts !

But indeed `re/src/sip/auth.c` currently supports only md5.
